packages/cli/src/workflows/workflows.controller.ts

```diff
@@ -355,6 +355,14 @@ workflowsController.patch(
 			);
 		}
 
+		if (incomingHash !== shared.workflow.hash) {
```

Comment:
Can we make this check only if the payload contains `nodes` and `connections`? Perhaps `pinData` too?

This reduces the chance of us making checks and warnings unnecessarily.

Reply:
https://github.com/n8n-io/n8n/pull/4446#discussion_r1006571003